platform/vs/docker-sonic-vs/start.sh

```diff
@@ -44,13 +47,39 @@ mkdir -p /etc/swss/config.d/
 
 rm -f /var/run/rsyslogd.pid
 
-supervisorctl start rsyslogd
+supervisord_cfg="/etc/supervisor/conf.d/supervisord.conf"
+chassis_cfg_file="/usr/share/sonic/hwsku/default_config.json"
+chassis_cfg_file_default="/etc/default/sonic-db/default_chassis_cfg.json"
+host_template="/usr/share/sonic/templates/hostname.j2"
+db_cfg_file="/var/run/redis/sonic-db/database_config.json"
+db_cfg_file_tmp="/var/run/redis/sonic-db/database_config.json.tmp"
+
+if [ -r "$chassis_cfg_file" ]; then
+   echo $(sonic-cfggen -j $chassis_cfg_file -t $host_template) >> /etc/hosts
+else
+   chassis_cfg_file="$chassis_cfg_file_default"
+   echo "10.8.1.200 redis_chassis.server" >> /etc/hosts
+fi
 
 mkdir -p /var/run/redis/sonic-db
 cp /etc/default/sonic-db/database_config.json /var/run/redis/sonic-db/
 
+supervisorctl reread
+supervisorctl update
+
+supervisorctl start rsyslogd
+
 supervisorctl start redis-server
 
+start_chassis_db=`sonic-cfggen -v DEVICE_METADATA.localhost.start_chassis_db -y $chassis_cfg_file`
+conn_chassis_db=`sonic-cfggen -v DEVICE_METADATA.localhost.connect_to_chassis_db -y $chassis_cfg_file`
+if [[ "$HOSTNAME" == *"supervisor"* ]] || [ "$start_chassis_db" == "1" ]; then
+   supervisorctl start redis-chassis
+   python /usr/bin/chassis_db.py
+elif [[ "$conn_chassis_db" != "1" ]]; then
+   remove_chassisdb_config -j $db_cfg_file
```

Comment:
I think I'm missing something here. Why do we need to delete configs from the database if the system isn't configured with a chassis DB? Why can't we only add these configs if there *is* a chassis DB?

Reply:
it is good to understand CHASSIS_DB configuration and convenient to change its configuration at run time, otherwise we have to look into the code for CHASSIS_DB configuration.
For real switch, we implemented the same logic, we need get agreement on how to handle this.

Reply:
That makes sense but I think I'm still missing something. On the line above this we check for a supervisor and check the conn_chassis_db field before setting everything up, so why is this removal step necessary? Is the `elif` case just for single-box setups (e.g. when it's not a SSI and also not an FSI)?

Reply:
let me refactor the code to make the logic clearer:

if start_chassis_db==1; then start redis_chassis service. fi
if start_chassis_db!=1 AND connect_chassis_db!=1; then remove_chassisdb_config; fi

The logic is we setup redis_chassis on SSI only, and remove chassisdb config for not a SSI AND not an FSI.
I think this should be clearer.